app/components/Analyst/ChangeStatus.tsx

```diff
@@ -68,112 +67,193 @@ const ModalDescription = ({ currentStatus, draftStatus }) => {
   );
 
+  const conditionalApproval =
+    conditionalApprovalDataByApplicationId?.edges[0]?.node;
   const [changeReason, setChangeReason] = useState('');
 
-  const [currentStatus, setcurrentStatus] = useState(
-    getStatus(analystStatus, statusTypes)
+  const [currentStatus, setCurrentStatus] = useState(
+    getStatus(status, statusTypes)
   );
   const [draftStatus, setDraftStatus] = useState(
-    getStatus(analystStatus, statusTypes)
+    getStatus(status, statusTypes)
   );
 
-  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
-    setDraftStatus(getStatus(e.target.value, statusTypes));
+  const conditionalApprovalData = conditionalApproval?.jsonData;
 
-    // Open modal using anchor tag
-    window.location.hash = '#change-status-modal';
-  };
+  // Check conditional approval requirements are met for external status change
+  const isAllowedConditionalApproval =
+    isExternalStatus &&
+    analystStatus === 'conditionally_approved' &&
+    conditionalApprovalData?.decision?.ministerDecision === 'Approved' &&
+    conditionalApprovalData?.response?.applicantResponse === 'Accepted';
+
+  const disabledStatuses =
+    disabledStatusList && disabledStatusList[currentStatus?.name];
 
-  if (analystStatus === 'withdrawn') {
+  useEffect(() => {
+    // update status when there is a relay store update
+    setCurrentStatus(getStatus(status, statusTypes));
+    setDraftStatus(getStatus(status, statusTypes));
+  }, [status]);
+
+  // No dropdown for withdrawn applications
+  if (status === 'withdrawn') {
     return <StyledWithdrawn>Withdrawn</StyledWithdrawn>;
   }
 
-  const handleSave = async () => {
+  const handleSave = async (value) => {
+    const statusInputName = isExternalStatus
+      ? `applicant_${value || draftStatus?.name}`
+      : draftStatus?.name;
+
     createStatus({
       variables: {
         input: {
           applicationStatus: {
-            applicationId,
+            applicationId: rowId,
             changeReason,
-            status: draftStatus.name,
+            status: statusInputName,
           },
         },
       },
       onCompleted: () => {
         setChangeReason('');
-        setcurrentStatus(draftStatus);
+        setCurrentStatus(draftStatus);
+      },
+      updater: (store) => {
+        store
+          .get(id)
+          .setValue(
+            statusInputName,
+            isExternalStatus ? 'externalStatus' : 'analystStatus'
+          );
       },
     });
   };
+  const handleChange = async (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setDraftStatus(getStatus(e.target.value, statusTypes));
+    const isAllowedExternalChange =
+      isExternalStatus && e.target.value === analystStatus;
+
+    if (!isExternalStatus) {
+      // open modal for internal status change
+      window.location.hash = `#change-status-modal`;
+    } else if (
+      // open modal for external status change
+      !isAllowedExternalChange ||
```

Comment:
I think it might be easier to read if we refactor this above. Something like:

```javascript
const isAllowedExternalChange = isExternalStatus && e.target.value === analystStatus
const isInvalidConditionalApproval = e.target.value === 'conditionally_approved' && !isAllowedConditionalApproval
...
if(!isAllowedExternalChange || isInvalidConditionalApproval){
  window.location...
}
```